packages/components/src/validated-form-controls/control-with-error.tsx

```diff
@@ -109,6 +111,58 @@ function UnforwardedControlWithError< C extends React.ReactElement >(
 		return () => validityTarget?.removeEventListener( 'invalid', handler );
 	}, [ getValidityTarget ] );
 
+	// Suppress the native error popover, while keeping the focus behavior intact.
+	useEffect( () => {
+		const validityTarget = getValidityTarget();
+
+		const supressNativePopover = ( event: Event ) => {
+			event.preventDefault();
+
+			const target = event.target as ValidityTarget;
+
+			if ( ! target.form ) {
+				target.focus();
+				return;
+			}
+
+			const firstErrorInForm = Array.from( target.form.elements ).find(
+				( el ) => ! ( el as ValidityTarget ).validity.valid
+			);
+
+			if ( firstErrorInForm === target ) {
+				target.focus();
+			}
```

Comment:
Could we use `:invalid` query selector to simplify this?


```suggestion
			if ( ! target.form || target.form.querySelector( ':invalid' ) === target ) {
				target.focus();
			}
```

Reply:
Ugh, this is a great idea I didn't think of, but unfortunately this `querySelector( ':invalid' )` seems to trigger a recursive validation check in jsdom and makes every test hang 😭 I reshuffled the original code to make it a bit simpler though (41d019f).